### ISSUE TITLE
The generated updateArgs have no update attribute

### ISSUE THREAD

**Issue by larryro** [prisma/prisma#19967]
### Bug description

When we define two models, one is xxx and the other is xxxUpdate, and xxxUpdate is placed before xxx, we cannot update xxx with Prisma Client.

![image](https://github.com/prisma/prisma/assets/5787117/625ad1b1-3134-41b3-8165-42ccd0b958a3)
![image](https://github.com/prisma/prisma/assets/5787117/5623dbf8-53ad-443a-8c54-90a88801f6b6)
![image](https://github.com/prisma/prisma/assets/5787117/5584b348-9816-4efc-8d0d-d18a30622c1f)

### How to reproduce

1. Define two models with no corresponding relationships, such as PostUpdate and Post and make sure PostUpdate is ahead of Post.

```
model PostUpdate {
  id    Int     @id @default(autoincrement())
  email String  @unique
  name  String?
}

model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
}

```
2. Use PrismaClient to update Post.

```
await prisma.post.update({
    where: { id: 1 },
    data: { published: true },
  });

```
3. We can see that where and data do not exist as parameter properties
![image](https://github.com/prisma/prisma/assets/5787117/43766619-3bbc-4062-a272-11de61a99c21)
![image](https://github.com/prisma/prisma/assets/5787117/97dbebfe-bdb7-494b-a6e0-4a214ef0886b)

### Expected behavior

We should be able to use the update method to update posts normally.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

model PostUpdate {
  id    Int     @id @default(autoincrement())
  email String  @unique
  name  String?
}

model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
}
```

```ts
await prisma.post.update({
    where: { id: 1 },
    data: { content: "test" },
  });
```

### Environment & setup

- OS: Windows
- Database:MySQL, SQLite
- Node.js version: v18.14.2

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 4.16.1
@prisma/client          : 4.16.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine b20ead4d3ab9e78ac112966e242ded703f4a052c (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli b20ead4d3ab9e78ac112966e242ded703f4a052c (at node_modules\@prisma\engines\migration-engine-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.16.0-66.b20ead4d3ab9e78ac112966e242ded703f4a052c
Default Engines Hash    : b20ead4d3ab9e78ac112966e242ded703f4a052c
Studio                  : 0.484.0
```

**Comment by FJ001** [comment]
Is it a system breaking bug?
Is xxxUpdate is being placed before xxx necessary in some cases?

**Comment by Jolg42** [comment]
I didn't try to reproduce this yet, but I think this is a type clash
Related issues
- https://github.com/prisma/prisma/issues/7518
- https://github.com/prisma/prisma/issues/9568
- https://github.com/prisma/prisma/issues/18902

@larryro Could you try renaming your `PostUpdate` model?

If you want, you can only change the Prisma Client name and keep the table name in the database the same using `@@map` like:
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

model PostSomething {
  id    Int     @id @default(autoincrement())
  email String  @unique
  name  String?
	@@map("PostUpdate")
}

model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
}
```

Let me know if that helped, we're aware of some type clashes, and we have ideas on how to fix them, I hope it happens "soon" but I can't say when.

**Comment by SevInf** [comment]
This issue is now fixed and the fix will be published as a part of `5.1` release. If you'd like to verify fix earlier, you can use dev snapshot `5.1.0-dev.24`. We don't recommend using it in production, but it should be enough to verify the fix.
